artemis-protocols/artemis-amqp-protocol/src/main/java/org/apache/activemq/artemis/protocol/amqp/proton/ProtonServerSenderContext.java

```diff
@@ -980,6 +983,19 @@ class DefaultController implements SenderController {
 
       @Override
       public Consumer init(ProtonServerSenderContext senderContext) throws Exception {
+         {
+            ProtonHandler handler;
+            Connection qpidConnection;
+            // Avoiding possible NPEs that could happen on mock tests
+            if (connection != null &&
+               (handler = connection.getHandler()) != null && (qpidConnection = handler.getConnection()) != null) {
+               if (qpidConnection.getRemoteState() == EndpointState.CLOSED) {
+                  logger.warn("AMQP Connection creating invalid consumer for closed connection", connection);
+                  throw new IllegalStateException("AMQP connection " + connection.getRemoteAddress() + " creating invalid consumer for closed connection");
+               }
+            }
+         }
+
```

Comment:
I'm not particularly fond of this "make mocking easier approach" as it ignore what should be an invalid case of missing proton handler and connection.  If a test needs to mock this it should be able to supply mocks that answer with mocked versions of these objects.

Reply:
fair enough... 

I"m in the process to making it ready to be merged now.. and I will cleanup this before I set it ready to merge


Thanks for looking into it.